src/js/_enqueues/lib/delayed-inline-script-loader.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * Handles delayed execution of before/after inline scripts for defer/async scripts.
+ *
+ * @output wp-includes/js/wp-delayed-inline-script-loader.js
+ */
+
+(function (window, document) {
+	var nonce = document.currentScript.nonce,
+		doneDependencies = new Set();
+
+	/**
+	 * Determines whether a script was loaded.
+	 *
+	 * @param {string} dep Dependency handle.
+	 * @returns {boolean} Whether dependency was done.
+	 */
+	function isDependencyDone(dep) {
+		return doneDependencies.has(dep);
+	}
+
+	/**
+	 * Runs an inline script.
+	 *
+	 * @param {HTMLScriptElement} script Script to run.
+	 */
+	function runInlineScript(script) {
+		var newScript;
+		script.dataset.wpDone = "1";
+		if (nonce && nonce !== script.nonce) {
+			console.error(
+				"CSP nonce check failed for after inline script. Execution aborted.",
+				script
+			);
+			return;
+		}
+		newScript = script.cloneNode(true);
+		newScript.type = "text/javascript";
+		script.parentNode.replaceChild(newScript, script);
+	}
+
+	/**
+	 * Runs the supplied inline scripts if all of their dependencies have been done.
+	 *
+	 * @param {NodeList<HTMLScriptElement>} scripts Scripts to run if ready.
+	 */
+	function runReadyInlineScripts(scripts) {
+		var i, len, deps;
+		for (i = 0, len = scripts.length; i < len; i++) {
+			deps = scripts[i].dataset.wpDeps.split(/,/);
+			if (deps.every(isDependencyDone)) {
+				runInlineScript(scripts[i]);
+			}
+		}
+	}
+
+	/**
+	 * Runs whenever a load event happens.
+	 *
+	 * @param {Event} event Event.
+	 */
+	function onScriptLoad(event) {
+		var matches, handle, script;
+		if (!(event.target instanceof HTMLScriptElement || event.target.id)) {
+			return;
+		}
+
+		// TODO: Consider adding a data attribute which specifically contains the handle.
+		matches = event.target.id.match(/^(.+)-js$/);
+		if (!matches) {
+			return;
+		}
+		handle = matches[1];
+		doneDependencies.add(handle);
+
+		/*
+		 * Return now if blocking because we cannot run delayed inline scripts because if we do, we could accidentally
+		 * run the before inline script for a dependent _before_ the after script of this blocking script is evaluated.
+		 */
+		if (!(event.target.async || event.target.defer)) {
+			return;
+		}
```

Comment:
Correct, we're not delaying inline scripts for blocking scripts. But this same `onScriptLoad` function runs for _all_ `load` events while the page is loading. And since a delayed script can have a blocking dependency, it is important for this condition to be here so that above the above logic can run:

```js
doneDependencies.add(handle)
```

This will ensure that when this function does run for a delayed script, it will be able to run the inline `before` script as it will be able to verify that all of its `data-wp-deps` are among `doneDependencies`.